compiler/src/iree/compiler/Codegen/Common/ConvertToDestinationPassingStylePass.cpp

```diff
@@ -253,6 +253,22 @@ static LogicalResult convertToDestinationPassingStyle(OpBuilder &b,
       [&](tensor::EmptyOp emptyTensorOp) -> WalkResult {
         for (auto result : emptyTensorOp->getResults()) {
           if (!result.getType().isa<RankedTensorType>()) continue;
+
+          // The tensor.empty op created by tiling unpack ops is intended to be
+          // a stack allocation. We can't replace it with destination buffer
+          // because it is larger than the destination buffer.
+          bool isUsedByNonPerfectUnpack = false;
```

Comment:
Could you post the IR that is obtained from the tiling of unpack.. I am wondering if we should just change `BufferizationAnalysis` to handle `UnPackOp` specially so that the `tensor.empty` automatically doesnt come into a equivalence set with a store. Then you dont need the special handling. If you post the IR, I can give you more pointers.

Reply:
I think it's hard to replace the empty op with `flow.tensor.load` stuff. The empty op could have more elements than the destination buffer. In any case, here is the dump before the pass: https://gist.github.com/hanhanW/d95b1c2c1f8a5e4b7b01c5737b42a78a/raw

Reply:
I see what you are saying. If you want to avoid some `tensor.empty` from being converted away, its probably better to do a pre-pass and convert `tensor.empty -> unpack` operation to `bufferization.alloc_tensor -> unpack` operation. This is a definitive hint to the bufferization infra to create an allocation. This also matches the intent of the `tensor.empty` and having split `bufferization.alloc_tensor` out of the `tensor.empty` semantics. The former is meant for real allocations, while latter is meant to just be a proxy for tensor types.

Reply:
Good point, thanks for the suggestion. This makes sense to me!